pymc_marketing/mmm/models/components/saturation.py

```diff
@@ -0,0 +1,334 @@
+    hill_saturation,
+    logistic_saturation,
+    michaelis_menten,
+)
+from pymc_marketing.mmm.utils import _get_distribution, _validate_model_config
+
+
+### SATURATION FUNCTIONS
+class BaseFunction:
+    def __init__(self, max_lagging, model):
+        self.max_lagging = max_lagging
+        self.model = model
+
+    def apply(self, data):
+        raise NotImplementedError("Subclasses must implement this method.")
+
+
+class HillSaturationComponent(BaseFunction):
+    """
+    A class representing the Hill Saturation component of a marketing mix model.
+
+    Parameters:
+    -----------
+    model : PyMC.Model
+        The PyMC model object.
+    model_config : dict
+        A dictionary containing the configuration parameters for the model.
+
+    Attributes:
+    -----------
+    REQUIRED_KEYS : list
+        A list of required keys in the model_config dictionary.
+
+    Methods:
+    --------
+    apply(data: Union[pm.Data, pm.MutableData]) -> pm.Deterministic:
+        Apply the Hill Saturation component to the given data.
+
+    """
+
+    REQUIRED_KEYS = [
+        "saturation_sigma",
+        "saturation_lambda",
+        "saturation_beta",
+    ]
+
+    def __init__(self, model, model_config):
+        self.model = model
+        self.model_config = model_config
+        _validate_model_config(
+            required_keys=self.REQUIRED_KEYS, model_config=self.model_config
+        )
+
+    def apply(self, data: Union[pm.Data, pm.MutableData]) -> pm.Deterministic:
+        """
+        Apply the Hill Saturation component to the given data.
+
+        Parameters:
+        -----------
+        data : Union[pm.Data, pm.MutableData]
+            The input data for the model.
+
+        Returns:
+        --------
+        pm.Deterministic
+            The deterministic variable representing the channel contributions.
+
+        """
+        self.saturation_sigma_dist = _get_distribution(
+            dist=self.model_config["saturation_sigma"]
+        )
+
+        self.saturation_lambda_dist = _get_distribution(
+            dist=self.model_config["saturation_lambda"]
+        )
+
+        self.saturation_beta_dist = _get_distribution(
+            dist=self.model_config["saturation_beta"]
+        )
+
+        with pm.modelcontext(self.model):
+            saturation_sigma = self.saturation_sigma_dist(
+                "saturation_sigma",
+                **self.model_config["saturation_sigma"]["kwargs"],
+                dims=("channel"),
+            )
+
+            # lam
+            saturation_lambda = self.saturation_lambda_dist(
+                "saturation_lambda",
+                **self.model_config["saturation_lambda"]["kwargs"],
+                dims=("channel"),
+            )
+
+            # beta
+            saturation_beta = self.saturation_beta_dist(
+                "saturation_beta",
+                **self.model_config["saturation_beta"]["kwargs"],
+            )
+
+            return pm.Deterministic(
```

Comment:
General comment for this return: This shouldn't be a Deterministic but the result will be wrapped in a deterministic in the MMM class method

Think this will save a lot of duplication and would be easier to for users to make their own Saturation for our model because this "channel_contributions" variable name is what is used in the plots. It shouldn't be flexible

Reply:
Agree, great catch, already apply it.